flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/utils/RexNodeExtractor.scala

```diff
@@ -271,6 +271,12 @@ object RexNodeExtractor extends Logging {
       }
     }
 
+    val inputRefFinder = new InputRefVisitor
+    predicate.accept(inputRefFinder)
+    // if no fields reached, it's not partition condition
+    if (inputRefFinder.getFields.length == 0) {
+      return false
+    }
     try {
```

Comment:
There is no need to new an `InputRefVisitor`. I think you need  to deal it in `isSupportedPartitionPredicate.visitor`. Changing the logic of `visitor.visitInputRef` and  override `visitRexCall` if needed.

Reply:
Good idea, I will fix this.